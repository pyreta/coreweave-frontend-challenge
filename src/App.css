header {
  text-align: center;
  margin-top: 30px;
  font-size: xx-large;
}