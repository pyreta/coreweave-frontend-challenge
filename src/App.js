import React, { useState } from "react";
import DataWidget from "./components/DataWidget";
import ArrayGenerator from "./components/ArrayGenerator";
import ArrayTransformer from "./components/ArrayTransformer";
import './App.css';

const App = () => {
  const [randomArray, setRandomArray] = useState([]);
  const [outputArray, setOutputArray] = useState([]);
  const [runtime, setRuntime] = useState(0);

  return (
    <div>
      <header>The Pointless App</header>
      <p>
        Generate a random string array. Then add as many transformations as you
        want. You can add duplicates of the same transformation. Transformations
        will happen in the order they are added.
      </p>
      <ArrayGenerator length={randomArray.length} setArray={setRandomArray} />
      <br></br>
      <ArrayTransformer
        inputArray={randomArray}
        outputArray={outputArray}
        setOutputArray={setOutputArray}
        setRuntime={setRuntime}
      />
      <br></br>
      <DataWidget label={"Output array length"} value={outputArray.length} />
      <DataWidget label={"Runtime (ms)"} value={Math.round(runtime)} />
    </div>
  );
};

export default App;