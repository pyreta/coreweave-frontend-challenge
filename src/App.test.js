import { render, screen } from '@testing-library/react';
import App from './App';

describe("App", () => {
  it('renders', () => {
    render(<App />);
    const linkElement = screen.getByText('The Pointless App');
    expect(linkElement).toBeInTheDocument();
  });

  it('match snapshot', () => {
    expect(render(<App />).container).toMatchSnapshot();
  });
});

