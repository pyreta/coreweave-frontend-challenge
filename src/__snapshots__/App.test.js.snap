// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App match snapshot 1`] = `
<div>
  <div>
    <header>
      The Pointless App
    </header>
    <p>
      Generate a random string array. Then add as many transformations as you want. You can add duplicates of the same transformation. Transformations will happen in the order they are added.
    </p>
    <p>
      Array not generated
    </p>
    <button>
      Generate random array
    </button>
    <label
      for="length"
    >
      of length:
    </label>
    <input
      max="10000"
      min="1"
      name="length"
      type="number"
    />
    <br />
    <div>
      Transformations
    </div>
    <div>
      <button
        text="Scramble each element"
      >
        Scramble each element
      </button>
      <button
        text="Swap case of each element"
      >
        Swap case of each element
      </button>
      <button
        text="Add random character to each element"
      >
        Add random character to each element
      </button>
      <button
        text="Remove elements with a capital B"
      >
        Remove elements with a capital B
      </button>
    </div>
    <ol>
      <div>
        No filters selected
      </div>
    </ol>
    <button>
      Apply changes to array
    </button>
    <br />
    <div>
      Output array length
      : 
      0
    </div>
    <div>
      Runtime (ms)
      : 
      0
    </div>
  </div>
</div>
`;
